sdks/go/pkg/beam/core/runtime/metricsx/metricsx.go

```diff
@@ -150,7 +183,15 @@ func extractGaugeValue(reader *bytes.Reader) (metrics.GaugeValue, error) {
 }
 
 func newLabels(miLabels map[string]string) *metrics.Labels {
-	labels := metrics.UserLabels(miLabels["PTRANSFORM"], miLabels["NAMESPACE"], miLabels["NAME"])
+	labels := metrics.Labels{}
+	if miLabels["PTRANSFORM"] != "" {
+		labels = metrics.UserLabels(miLabels["PTRANSFORM"], miLabels["NAMESPACE"], miLabels["NAME"])
+		return &labels
+	}
+	if miLabels["PCOLLECTION"] != "" {
+		labels = metrics.PCollectionLabels(miLabels["PCOLLECTION"])
+		return &labels
+	}
 	return &labels
```

Comment:
Unless one is setting fields conditionally, there's no reason to pre-declare a variable at the top of a function.
```suggestion
	if miLabels["PTRANSFORM"] != "" {
		labels := metrics.UserLabels(miLabels["PTRANSFORM"], miLabels["NAMESPACE"], miLabels["NAME"])
		return &labels
	}
	if miLabels["PCOLLECTION"] != "" {
		labels := metrics.PCollectionLabels(miLabels["PCOLLECTION"])
		return &labels
	}
	return &metrics.Labels{}
```